classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -36,79 +31,154 @@ protected function getContextKeyField(): string
-            }
-
-            // Clean orphaned completed_payments entries by user_id
-            $orphanedIds = DB::table('completed_payments AS cp')->leftJoin('users AS u', 'u.user_id', '=', 'cp.user_id')->whereNull('u.user_id')->distinct()->pluck('cp.user_id');
-            foreach ($orphanedIds as $userId) {
-                DB::table('completed_payments')->where('user_id', '=', $userId)->delete();
-            }
-        } catch (\Exception $e) {
-            if ($fallbackVersion = $this->setFallbackVersion()) {
-                $this->_installer->log("A pre-flight check failed. The software was successfully upgraded to {$fallbackVersion} but could not be upgraded further (to " . $this->_installer->newVersion->getVersionString() . '). Check and correct the error, then try again.');
-            }
-            throw $e;
-        }
+        parent::buildOrphanedEntityProcessor();
+
+        $this->addTableProcessor('publication_formats', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~2 entities: doi_id->dois.doi_id(not found in previous version) publication_id->publications.publication_id
+            $affectedRows += $this->deleteRequiredReference('publication_formats', 'publication_id', 'publications', 'publication_id');
+            return $affectedRows;
+        });
+
+        $this->addTableProcessor('submission_chapters', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~4 entities: doi_id->dois.doi_id(not found in previous version) primary_contact_id->authors.author_id publication_id->publications.publication_id source_chapter_id->submission_chapters.chapter_id
+            $affectedRows += $this->deleteRequiredReference('submission_chapters', 'publication_id', 'publications', 'publication_id');
+
+            $affectedRows += $this->cleanOptionalReference('submission_chapters', 'primary_contact_id', 'authors', 'author_id');
+            return $affectedRows;
+        });
+
+        $this->addTableProcessor('publications', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~4 entities: primary_contact_id->authors.author_id doi_id->dois.doi_id(not found in previous version) series_id->series.series_id submission_id->submissions.submission_id
+            $affectedRows += $this->cleanOptionalReference('publications', 'series_id', 'series', 'series_id');
+            return $affectedRows;
+        });
+
+        $this->addTableProcessor('series', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~2 entities: press_id->presses.press_id review_form_id->review_forms.review_form_id
+            $affectedRows += $this->deleteRequiredReference('series', $this->getContextKeyField(), $this->getContextTable(), $this->getContextKeyField());
+            $affectedRows += $this->cleanOptionalReference('series', 'review_form_id', 'review_forms', 'review_form_id');
```

Comment:
The previous migration was deleting these records, but as "sections" are only required on OJS, I changed the behavior.